packages/block-editor/src/store/selectors.js

```diff
@@ -1576,7 +1575,21 @@ export function getTemplateLock( state, rootClientId ) {
 		return state.settings.templateLock ?? false;
 	}
 
-	return getBlockListSettings( state, rootClientId )?.templateLock ?? false;
+	const blockListTemplateLock = getBlockListSettings(
+		state,
+		rootClientId
+	)?.templateLock;
+
+	// If this is a contentOnly template locked block that's being in the process
```

Comment:
Another tiny optional nit (feel free to ignore!)

```suggestion
	// If this is a contentOnly template locked block that's in the process
```